File: android/SagaiSambaandh/app/src/main/java/com/sagaisambaandh/data/DataModel.kt
package com.sagaisambaandh.data

data class Profile(
    val id: String,
    val name: String,
    val age: Int,
    val gender: String, // "Bride" or "Groom"
    val clan: String,
    val gotra: String,
    val kul: String,
    val thikana: String,
    val location: String,
    val height: String,
    val occupation: String,
    val education: String,
    val income: String,
    val isVerified: Boolean,
    val img: String? = null // local resource identifier
)

data class Clan(
    val name: String,
    val origin: String,
    val dynasty: String,
    val history: String
)

data class User(
    val id: String,
    val name: String,
    val email: String,
    val gender: String,
    val clan: String,
    val tier: String = "Starter", // "Starter", "Silver", "Gold"
    val shortlistedIds: Set<String> = emptySet(),
    val unlockedIds: Set<String> = emptySet()
)

File: SagaiSambaandh/SagaiSambaandh/ios/SagaiSambaandh/DataModel.swift
import Foundation

struct Profile: Identifiable, Codable, Hashable {
    var id: String
    var name: String
    var age: Int
    var gender: String // "Bride" or "Groom"
    var clan: String
    var gotra: String
    var kul: String
    var thikana: String
    var location: String
    var height: String
    var occupation: String
    var education: String
    var income: String
    var isVerified: Bool
    var img: String? // Image file reference
}

struct Clan: Identifiable, Codable, Hashable {
    var id: String { name }
    var name: String
    var origin: String
    var dynasty: String
    var history: String
}

struct User: Identifiable, Codable, Hashable {
    var id: String
    var name: String
    var email: String
    var gender: String // "Groom" or "Bride"
    var clan: String
    var tier: String // "Starter", "Silver", or "Gold"
    var shortlistedIds: [String]
    var unlockedIds: [String]
    var gotra: String = ""
    var motherGotra: String = ""
    var thikana: String = ""
    var phone: String = ""
    var dob: String = ""
    var education: String = ""
    var occupation: String = ""
    var income: String = ""
    var height: String = ""
    var maritalStatus: String = "Never Married"
    var profilePic: String? = nil
    var isNewUser: Bool? = false
}
